docker-installer/install_cni_plugins.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+set -e
+
+if [ $# -ne 2 ]; then
+  echo "USAGE: $0 source-dir dest-dir"
+  exit 1
+fi
+
+SRC=$1
+DST=$2
+
+install_cni_plugin() {
+	NAME=$1
+	MD5=$( md5sum $SRC/$NAME | awk '{ print $1 }' )
+	if [ -e $DST/$NAME ]; then
+		if [ ! -e $DST/$NAME.by_cni_installer_image ]; then
+			# The file already exists but there's no marker that
+			# it was installed by this installer -> keep untouched
+			echo "* '$NAME' ignored (exists but not installed by me)"
+			return
+		fi
+		OTHER_MD5=$( md5sum $DST/$NAME | awk '{ print $1 }' )
+		INSTALLED_MD5=$( cat $DST/$NAME.by_cni_installer_image )
+
+		if [ "$OTHER_MD5" != "$INSTALLED_MD5" ]; then
+			# The file was previously installed by this installer
+			# but later changed -> keep untouched
+			echo "* '$NAME' ignored (previously installed by me but changed by someone else)"
+			return
+		fi
+
+		if [ "$OTHER_MD5" == "$MD5" ]; then
+			# The file was previously installed by this installer
+			# and is up-to-date
+			echo "* '$NAME' is up-to-date"
+			return
+		fi
+
+		# The file was previously installed by this installer
+		# but needs an update
+		cp -a $SRC/$NAME $DST/$NAME
```

Comment:
@micw do we really want `-a` here? that adds `-R` and I don't think we want recursive copy of a single file right?

Reply:
I'm impressed that my PR woke up after 2 years of deep sleep ;-)
If I recall correctly, the plug-ins are executables, so either -a is required or a chmod/chown afterwards.

Reply:
@micw `-a` is an alias for `-dR --preserve=all` but we dont' want the `-R`. So you could change this to:

`cp -d --preserve=all $SRC/$NAME $DST/$NAME`

and solve the issue?